### PR TITLE
Generate base seed for DataLoaderIter only if num workers larger than 0

### PR DESCRIPTION
**Pull request by timmeinhardt**
This PR moves the generation of the workers `base_seed` such that it is only generated if data is fetched with workers, i.e, `num_workers > 0`. This allows to run for example a validation set without changing the random state und ensuring determinism in between to different experiments in which the number of validation times differs.

### CODE REVIEW

**Review by ssnl — CHANGES_REQUESTED** [review]
This is specifically coded this way for reproducibility regardless of `num_workers`

**Comment by timmeinhardt** [comment]
I see, but I find it quite unintuitive that a validation set run without shuffle on a model in evaluation mode changes the random state. Could the base seed be generated during the initialisation of the `Dataloader` or do you need a different seed for each `Dataloader.__iter__` call?

**Comment by SsnL** [comment]
@timmeinhardt IMO, one shouldn't rely on assumption that RNG state shouldn't change during some code. It would be much better if you just save the state before the val run, and load it afterwards. Using seeds also works.

**Comment by timmeinhardt** [comment]
Okay, that makes sense. Thank you for your support!
